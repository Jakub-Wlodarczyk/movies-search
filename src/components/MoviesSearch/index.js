import React from 'react';
import {
	getMovie,
	getConfiguration
} from '../../app/resources';
import movieDbLogo from '../../assets/movie_db_logo.png';
import MoviesList from '../MoviesList';

/**
 * The function provides a movie search functionality
 * @returns {JSX}
 */
class MoviesSearch extends React.Component {
	constructor (props) {
		super(props);

		this.state = {
			searchPhrase: '',
			searchInfo: '',
			movies: [],
			APIImageBaseUrl: '',
			isSearchActive: true
		};

		this.handleSearchChange = this.handleSearchChange.bind(this);
		this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
	}

	componentDidMount () {
		getConfiguration().then(config => this.setState({
			APIImageBaseUrl: config.images.base_url
		}));
	}

	handleSearchChange (e) {
		this.setState({
			searchPhrase: e.target.value
		});
	}

	handleSearchSubmit (e) {
		e.preventDefault();
		const searchPhrase = this.state.searchPhrase;

		getMovie(searchPhrase).then(data => {
			let movies = [];
			let searchMessage;
			let isSearchActive;

			if (data.errors) {
				searchMessage = data.errors[0];
				isSearchActive = true;
			} else if (data.total_results === 0) {
				searchMessage = 'There were no results for your search. Please try with a different title.';
				isSearchActive = true;
			} else {
				movies = data.results;
				searchMessage = '';
				isSearchActive = false;
			}

			this.setState({
				movies,
				searchInfo: searchMessage,
				isSearchActive
			})
		});
	}

	render () {
		return (
			<div className="movies-search">
				<div className="movies-search__movie-db-logo">
					<img name="movie-db-logo-image" src={movieDbLogo} />
				</div>
				{this.state.isSearchActive &&
				<form onSubmit={this.handleSearchSubmit} name='movies-search-form'>
					<label>
						Search for a movie:
						<input
							type="text"
							name="searchPhrase"
							placeholder="example movie: Shrek"
							value={this.state.searchPhrase}
							onChange={this.handleSearchChange}
						/>
					</label>
					<input type="submit" value="Search"/>
				</form>
				}

				{this.state.searchInfo.length > 0 &&
				<p className="movies-search__search-info">{this.state.searchInfo}</p>
				}

				{this.state.movies.length > 0 &&
					<MoviesList movies={this.state.movies} baseUrl={this.state.APIImageBaseUrl} />
				}
			</div>
		);
	}
}

export default MoviesSearch;
