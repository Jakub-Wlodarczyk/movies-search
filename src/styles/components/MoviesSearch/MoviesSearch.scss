.movies-search__search-info {
  color: #da2121;
  font-weight: bold;
}

.movies-search__movie-db-logo {
  position: absolute;
  right: 0;
  top: 0;
  width: 80px;
  height: 80px;

  img {
    width: 80px;
    height: 80px;
  }
}