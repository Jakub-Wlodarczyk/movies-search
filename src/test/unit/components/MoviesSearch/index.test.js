import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MoviesSearch from '../../../../components/MoviesSearch';

describe('MoviesSearch', () => {
	it('should render the MoviesSearch view', () => {
		const wrapper = shallow(<MoviesSearch />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render the MoviesSearch container', () => {
		const wrapper = shallow(<MoviesSearch />);
		expect(wrapper.find('.movies-search').exists()).toBeTruthy();
	});

	it('should render the MoviesSearch search form', () => {
		const wrapper = shallow(<MoviesSearch />);
		expect(wrapper.find('form[name="movies-search-form"]').exists()).toBeTruthy();
	});
});
