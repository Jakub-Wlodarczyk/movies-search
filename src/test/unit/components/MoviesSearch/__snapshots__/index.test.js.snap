// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSearch should render the MoviesSearch view 1`] = `
<div
  className="movies-search"
>
  <div
    className="movies-search__movie-db-logo"
  >
    <img
      name="movie-db-logo-image"
      src=""
    />
  </div>
  <form
    name="movies-search-form"
    onSubmit={[Function]}
  >
    <label>
      Search for a movie:
      <input
        name="searchPhrase"
        onChange={[Function]}
        placeholder="example movie: Shrek"
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Search"
    />
  </form>
</div>
`;
